modules/tasks/tasks/src/main/java/org/motechproject/tasks/service/impl/TaskActivityServiceImpl.java

```diff
@@ -30,64 +37,120 @@ public TaskActivityServiceImpl(TaskActivitiesDataService taskActivitiesDataServi
     }
 
     @Override
-    public void addError(Task task, TaskHandlerException e, Map<String, Object> parameters) {
-        taskActivitiesDataService.create(new TaskActivity(e.getMessage(), e.getArgs(), task.getId(),
-                TaskActivityType.ERROR, ExceptionUtils.getStackTrace(e), parameters));
+    @Transactional(propagation = Propagation.REQUIRES_NEW)
```

Comment:
No. What Tasks module needs to do after successful or failed execution is to update the ->existing<- activity. Since the task actions can be either events or service calls, it might happen that some actions (event-based) are executed even before the Tasks module is done starting other actions. In that case, the executed actions would fail to update their respective activity, since they are not commited in the database yet. Creating initial task activity is done right before executing actions (in the same transaction, if we do not start new one here).